src/composables/useEmailSubscription.ts

```diff
@@ -1,70 +1,91 @@
-import sign, { SubscribeType } from '@/helpers/sign';
-import { getInstance } from '@snapshot-labs/lock/plugins/vue3';
+import { createSharedComposable } from '@vueuse/core';
+import { useEmailFetchClient } from './useEmailFetch';
 
-export function useEmailSubscription() {
-  enum Status {
-    waiting,
-    success,
-    error
-  }
+enum Status {
+  waiting,
+  success,
+  error
+}
 
-  enum Level {
-    info = 'info',
-    warning = 'warning',
-    'warning-red' = 'warning-red',
-    success = 'success'
-  }
+enum Level {
+  info = 'info',
+  warning = 'warning',
+  'warning-red' = 'warning-red',
+  success = 'success'
+}
+
+const subscriptionTypes = ['summary', 'newProposal', 'closedProposal'] as const;
+type SubscriptionType = (typeof subscriptionTypes)[number];
+
+function useEmailSubscriptionComposable() {
+  const { t } = useI18n();
 
   const status: Ref<Status> = ref(Status.waiting);
```

Comment:
In Vue 3, `ref<Type>(value)` is used to create a reactive reference with an initial value, where `Type` is the TypeScript type of `value`. This provides better type inference than `const status: Ref<Status> = ref(Status.waiting);` because it ensures that `status.value` is of `Status` type, instead of `status` itself being `Status`.